packages/iris-grid/src/sidebar/TableCsvExporter.tsx

```diff
@@ -184,12 +186,24 @@ class TableCsvExporter extends Component<
     const snapshotRanges = this.getSnapshotRanges();
     if (this.validateOptionInput()) {
       onDownloadStart();
-      model.export().then(frozenTable => {
+      try {
+        const frozenTable = await model.export();
         const tableSubscription = frozenTable.setViewport(0, 0);
-        tableSubscription.getViewportData().then(() => {
-          onDownload(fileName, frozenTable, tableSubscription, snapshotRanges);
+        await tableSubscription.getViewportData();
+        onDownload(fileName, frozenTable, tableSubscription, snapshotRanges);
+      } catch (error) {
+        log.error('CSV download failed', error);
+
+        this.setState({
+          errorMessage: (
+            <p>
+              <FontAwesomeIcon icon={vsWarning} /> CSV download failed,
+              TableCsvExporter rejected promise.
```

Comment:
This is a user facing message that we are displaying. Instead of displaying `TableCsvExporter rejected promise.`, which most users have no idea what that means, we should print the error itself, which we need to stringify to get it to appear correctly, e.g.:
```suggestion
              <FontAwesomeIcon icon={vsWarning} /> {`${error}`}
```